From 1dbc086c64471eb7d4910898408107bbd930ccc4 Mon Sep 17 00:00:00 2001
From: "@KoboDev" <KoboDev@mail.com>
Date: Sun, 7 Feb 2021 11:19:47 -0600
Subject: [PATCH] Fixed pom.xml to an live link, additionally fixed an error
 with craftserver.java that has an IOException instead of a regular exception.


diff --git a/dependency-reduced-pom.xml b/dependency-reduced-pom.xml
new file mode 100644
index 00000000..6c10d05f
--- /dev/null
+++ b/dependency-reduced-pom.xml
@@ -0,0 +1,117 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <parent>
+    <artifactId>canyon-parent</artifactId>
+    <groupId>com.canyonmodded</groupId>
+    <version>1.7.3</version>
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+  <artifactId>canyon-server</artifactId>
+  <name>Canyon-Server</name>
+  <version>1.7.3</version>
+  <url>http://www.bukkit.org</url>
+  <scm>
+    <connection>scm:git:git://github.com/Bukkit/CraftBukkit.git</connection>
+    <developerConnection>scm:git:ssh://git@github.com/Bukkit/CraftBukkit.git</developerConnection>
+    <url>https://github.com/Bukkit/CraftBukkit</url>
+  </scm>
+  <build>
+    <defaultGoal>clean install</defaultGoal>
+    <plugins>
+      <plugin>
+        <groupId>net.md-5</groupId>
+        <artifactId>scriptus</artifactId>
+        <version>0.3.1</version>
+        <executions>
+          <execution>
+            <phase>initialize</phase>
+            <goals>
+              <goal>describe</goal>
+            </goals>
+          </execution>
+        </executions>
+        <configuration>
+          <format>git-Canyon-%s</format>
+        </configuration>
+      </plugin>
+      <plugin>
+        <artifactId>maven-jar-plugin</artifactId>
+        <version>2.1</version>
+        <configuration>
+          <archive>
+            <manifestEntries>
+              <Main-Class>org.bukkit.craftbukkit.Main</Main-Class>
+              <Implementation-Title>Canyon</Implementation-Title>
+              <Implementation-Version>${describe}</Implementation-Version>
+              <Implementation-Vendor>Canyon Team</Implementation-Vendor>
+              <Specification-Title>Bukkit</Specification-Title>
+              <Specification-Version>${api.version}</Specification-Version>
+              <Specification-Vendor>Bukkit Team</Specification-Vendor>
+              <Sealed>true</Sealed>
+            </manifestEntries>
+            <manifestSections>
+              <manifestSection>
+                <name>net/bukkit/</name>
+                <manifestEntries>
+                  <Sealed>true</Sealed>
+                </manifestEntries>
+              </manifestSection>
+              <manifestSection>
+                <name>com/bukkit/</name>
+                <manifestEntries>
+                  <Sealed>true</Sealed>
+                </manifestEntries>
+              </manifestSection>
+              <manifestSection>
+                <name>org/bukkit/</name>
+                <manifestEntries>
+                  <Sealed>true</Sealed>
+                </manifestEntries>
+              </manifestSection>
+            </manifestSections>
+          </archive>
+        </configuration>
+      </plugin>
+      <plugin>
+        <artifactId>maven-shade-plugin</artifactId>
+        <version>3.2.1</version>
+        <executions>
+          <execution>
+            <phase>package</phase>
+            <goals>
+              <goal>shade</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <version>3.8.1</version>
+        <configuration>
+          <source>1.8</source>
+          <target>1.8</target>
+        </configuration>
+      </plugin>
+    </plugins>
+  </build>
+  <repositories>
+    <repository>
+      <id>bukkit-repo</id>
+      <url>https://download938.mediafire.com/jzs3lz3gm1jg/xuiz2lwjyfiljfp/minecraft-server.jar</url>
+    </repository>
+    <repository>
+      <id>canyon-local-repo</id>
+      <url>file://${basedir}/../work/local-repo</url>
+    </repository>
+  </repositories>
+  <pluginRepositories>
+    <pluginRepository>
+      <id>bukkit-plugins</id>
+      <url>https://hub.spigotmc.org/nexus/content/repositories/central/</url>
+    </pluginRepository>
+  </pluginRepositories>
+  <properties>
+    <api.version>unknown</api.version>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+  </properties>
+</project>
diff --git a/pom.xml b/pom.xml
index 0760f87c..b0c60311 100644
--- a/pom.xml
+++ b/pom.xml
@@ -24,7 +24,7 @@
   <repositories>
     <repository>
       <id>bukkit-repo</id>
-      <url>https://repo.canyonmodded.com</url> <!-- Canyon: replace repo -->
+      <url>https://download938.mediafire.com/jzs3lz3gm1jg/xuiz2lwjyfiljfp/minecraft-server.jar</url> <!-- Canyon: replace repo -->
     </repository>
     <repository>
       <id>canyon-local-repo</id>
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 80ce92ca..17df940c 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -494,7 +494,7 @@ public final class CraftServer implements Server {
                 getLogger().info("Converting world '" + name + "'");
                 converter.convert(name, new ConvertProgressUpdater(console));
             }
-        } catch (IOException e) {
+        } catch (Exception e) {
             // launder the exception
             throw new RuntimeException(e);
         }
-- 
2.30.0.windows.2

